pkg/runner/test_runner.go

```diff
@@ -471,13 +400,37 @@ loop:
 		// Make sure we have a step runner active. If not, start one.
 		err := tr.runStepIfNeeded(ss)
 		// Inject the target.
-		if err == nil && inject {
-			err = tr.injectTarget(ctx, tgs, ss)
+		var targetResult ResultNotifier
```

Comment:
nit; the notify aspect is important for these objects
```suggestion
		var targetResultNotif ResultNotifier
```
or targetNotifier

same for other places where this naming is used